src/Grpc.AspNetCore.Server/Internal/CallHandlers/ServerCallHandlerBase.cs

```diff
@@ -51,32 +51,12 @@ public Task HandleCallAsync(HttpContext httpContext)
         {
             if (GrpcProtocolHelpers.IsInvalidContentType(httpContext, out var error))
             {
-                // This might be a CORS preflight request and CORS middleware hasn't been configured
-                if (GrpcProtocolHelpers.IsCorsPreflightRequest(httpContext))
-                {
-                    GrpcServerLog.UnhandledCorsPreflightRequest(Logger);
-
-                    GrpcProtocolHelpers.BuildHttpErrorResponse(httpContext.Response, StatusCodes.Status405MethodNotAllowed, StatusCode.Internal, "Unhandled CORS preflight request received. CORS may not be configured correctly in the application.");
-                    httpContext.Response.Headers[HeaderNames.Allow] = HttpMethods.Post;
-                    return Task.CompletedTask;
-                }
-                else
-                {
-                    GrpcServerLog.UnsupportedRequestContentType(Logger, httpContext.Request.ContentType);
-
-                    GrpcProtocolHelpers.BuildHttpErrorResponse(httpContext.Response, StatusCodes.Status415UnsupportedMediaType, StatusCode.Internal, error);
-                    return Task.CompletedTask;
-                }
+                return ProcessInvalidContentTypeRequest(httpContext, error);
```

Comment:
Why extract this method? Seems unnecessary.

Reply:
Just splitting the existing method up into smaller parts